### PR TITLE
Added an option to customise the VDB namespace

### PR DESCRIPTION
**Pull request by Idclip**
Signed-off-by: Nick Avramoussis <4256455+Idclip@users.noreply.github.com>

### CODE REVIEW

**Comment by Idclip** [comment]
@jmlait I added this as a separator between the two versions in an attempt to help match example usage of this (Houdini) but it looks like Houdini suffixes the namespace with `__sesi` in the first instance and `__sesi__` in the second instance, so not sure this change actually helps you if you want to keep those (though `OPENVDB_REQUIRE_VERSION_NAME` is off by default). As such it might make more sense if this is simply `OPENVDB_NAMESPACE_SUFFIX`.

File: openvdb/openvdb/version.h.in [hunk 1]
```diff
@@ -111,9 +111,9 @@
 ///
 /// where X, Y and N are the major, minor and ABI version numbers, respectively.
 #if OPENVDB_ABI_VERSION_NUMBER == OPENVDB_LIBRARY_MAJOR_VERSION_NUMBER
-    #define OPENVDB_VERSION_NAME v${OpenVDB_MAJOR_VERSION}_${OpenVDB_MINOR_VERSION}
+    #define OPENVDB_VERSION_NAME v${OpenVDB_MAJOR_VERSION}_${OpenVDB_MINOR_VERSION}${OPENVDB_CUSTOM_NAMESPACE}
 #else
-    #define OPENVDB_VERSION_NAME v${OpenVDB_MAJOR_VERSION}_${OpenVDB_MINOR_VERSION}abi${OPENVDB_ABI_VERSION_NUMBER}
+    #define OPENVDB_VERSION_NAME v${OpenVDB_MAJOR_VERSION}_${OpenVDB_MINOR_VERSION}${OPENVDB_CUSTOM_NAMESPACE}abi${OPENVDB_ABI_VERSION_NUMBER}
```
Comment (danrbailey) [inline openvdb/openvdb/version.h.in:116]: Why put it before the abi?

**Review by danrbailey — COMMENTED** [review]
> @jmlait I added this as a separator between the two versions in an attempt to help match example usage of this (Houdini) but it looks like Houdini suffixes the namespace with `__sesi` in the first instance and `__sesi__` in the second instance, so not sure this change actually helps you if you want to keep those (though `OPENVDB_REQUIRE_VERSION_NAME` is off by default). As such it might make more sense if this is simply `OPENVDB_NAMESPACE_SUFFIX`.

I don't really follow this comment - what is first and second instances? I like `OPENVDB_NAMESPACE_SUFFIX` though, but would prefer the "_" to be introduced by the macro so this can just be a single word such as "sesi".

**Comment by Idclip** [comment]
Not sure I agree about auto adding the _, this way is far more explicit - but can change if you feel it's necessary. The first and second instance refers to whether the abi is suffixed or not. In sesis case, v8_1_sesi vs v8_1_sesi_abi7

**Comment by danrbailey** [comment]
> Not sure I agree about auto adding the _, this way is far more explicit - but can change if you feel it's necessary. The first and second instance refers to whether the abi is suffixed or not. In sesis case, v8_1_sesi vs v8_1_sesi_abi7

If there's already precedence of this pattern being used in Houdini for abi releases, that's fine, I just wasn't aware this was being done.

I have a soft preference for enforcing the underscore, but it's not a deal breaker.

**Review by danrbailey — APPROVED** [review]

**Comment by Idclip** [comment]
> If there's already precedence of this pattern being used in Houdini for abi releases, that's fine, I just wasn't aware this was being done.

If @jmlait doesn't have a preference then I'd also it rather be a suffix.

**Review by jmlait — APPROVED** [review]
I don' t have a preference, I don't think we need the namespace to be a specific pattern, so long as we can make it different than the shipped.

**Comment by Idclip** [comment]
I've changed this to be a suffix

**Pull request merged**